src/careamics/prediction_utils/stitch_prediction.py

```diff
@@ -76,8 +76,22 @@ def stitch_prediction_single(
     numpy.ndarray
         Full image, with dimensions SC(Z)YX.
     """
-    # retrieve whole array size
-    input_shape = (1, *tile_infos[0].array_shape)  # add S dim
+    # TODO: this is hacky... need a better way to deal with when input channels and
+    #   target channels do not match
+    if len(tile_infos[0].array_shape) == 4:
+        # 4 dimensions => 3 spatial dimensions so -4 is channel dimension
+        tile_channels = tiles[0].shape[-4]
+    elif len(tile_infos[0].array_shape) == 3:
+        # 3 dimensions => 2 spatial dimensions so -3 is channel dimension
+        tile_channels = tiles[0].shape[-3]
+    else:
```

Comment:
I guess this could be simplified to (at least) )something like 
` if len(tile_infos[0].array_shape) >=3:
    tile_channels = tiles[0].shape[-len(tile_infos[0].array_shape)] 
   else:
       error `
tile_channels = tiles[0].shape[

Reply:
`TileInformation.array_shape` is a tuple so it does not support element assignment. In any case this should not happen here and the array shape should be correct before reaching this function. This should be fixed in the dataset classes.